web_external/js/views/widgets/AddS3SourceWidget.js

```diff
@@ -0,0 +1,46 @@
+/**
+* This widget displays a form for adding a new S3 source.
+*/
+minerva.views.AddS3SourceWidget = minerva.View.extend({
+
+    events: {
+        'submit #m-add-s3-source-form': function (e) {
+            e.preventDefault();
+
+            var params = {
+                name: $('#m-s3-source-name').val(),
+                bucket: $('#m-s3-bucket').val(),
+                prefix: $('#m-s3-prefix').val(),
+                accessKeyId: $('#m-s3-access-key-id').val(),
+                secret: $('#m-s3-secret').val(),
+                service: $('#m-s3-service').val(),
+                readOnly: $('#m-s3-readonly').val(),
+                folderId: this.collection.folderId
+            };
+
+            if (params.prefix.indexOf('/') !== 0) {
+                params.prefix = '/' + params.prefix;
+            }
+
```

Comment:
Can we do some primitive client-side validation here similar to [this](https://github.com/Kitware/minerva/blob/bsve_analysis_update/web_external/js/views/widgets/ElasticsearchWidget.js#L24)?